doc/man/scons.xml

```diff
@@ -4654,70 +4667,109 @@ env = conf.Finish()
-some variables need to be specified at build time.
-For example, libraries needed for the build may be in non-standard
-locations, or site-specific compiler options may need to be passed to the
-compiler.
-&SCons;
-provides a <firstterm>&Variables;</firstterm>
-object to support overriding &consvars; with values obtained
-from various sources, often from the command line:</para>
+specialized information needs to be conveyed at build time
+to override the defaults in the build scripts.
+Command-line arguments (like <literal>--implcit-cache</literal>)
+and giving names of build targets are two ways to do that.
+Another is to provide variable-assignment arguments
+on the command line.
+For the particular case where you want to specify new
+values for &consvars;,
+&SCons; provides a <firstterm>&Variables;</firstterm>
+object to simplify collecting those
+and updating a &consenv; with the values.
+The typical calling style looks like:
+</para>
 
 <screen>
 <userinput>scons VARIABLE=foo</userinput>
 </screen>
 
-<para>The variable values can also be specified in a configuration file
-or an &SConscript; file.</para>
+<para>
+Variables specified in the above way
+can be manually processed by accessing the
+<link linkend="v-ARGUMENTS">&ARGUMENTS;</link> dictionary
+(or <link linkend="v-ARGLIST">&ARGLIST;</link> list),
+but using a &Variables; object allows you to describe
+anticipated variables,
+convert them to a suitable type if necessary,
+validate the values are within defined constraints,
+and define defaults, help messages and aliases.
+This is conceptually similar to the structure of options
+(see &f-link-AddOption;).
+It also allows obtaining values from a saved variables file,
+or from a custom dictionary in an &SConscript; file.
+The processed variables can then be applied to the desired &consenv;.
+</para>
+
+<para>
+Roughly speaking, arguments are used to convey information to the
+&SCons; program about how it should behave;
+variables are used to convey information to the build
+(although &SCons; does not enforce any such constraint).
+</para>
 
-<para>To obtain the object for manipulating values,
-call the &Variables; function:</para>
+<para>To obtain an object for manipulating variables,
+call the &Variables; factory function:</para>
 
 <variablelist>
   <varlistentry id="v-Variables">
   <term><function>Variables</function>([<parameter>files, [args]]</parameter>)</term>
   <listitem>
-<para>If <parameter>files</parameter> is a file or
-list of files, they are executed as &Python; scripts,
-and the values of (global) &Python; variables set in
-those files are added as &consvars; in the &DefEnv;.
-If no files are specified,
+<para>If <parameter>files</parameter> is a filename or list of filenames,
+they are considered to be &Python; scripts which will
+be executed to set variables when the
+<link linkend='v-Update'><function>Update</function></link>
+method is called -
+this allows the use of &Python; syntax in the assignments.
```

Comment:
A review question from the author: this was never stated explicitly. It's an artifact of how the file is processed (` exec(contents, {}, values) `). We could continue to avoid saying so (and remove from the example again).  I ask because one could switch the implementation to use the `tomllib` reader, which consumes Python-like assignment statements, and stop doing `exec` (which provides a possible attack vector against an SCons build). But if it's documented that it's Python *code* rather than just simple assignment statements, that would be a harder change to make.

Reply:
Okay, I was inaccurate, current docs do say "executed as a Python script" (manpage and user guide both), although it is not more explicit about what that means. I guess that horse left the stable long ago...